<!DOCTYPE html>
<html lang="es">
<head>
    <meta charset="utf-8">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Bloque Try-Catch</title>

    <link rel="alternate" href="https://campusempresa.com/mod/java/06-02-try-catch-block" hreflang="es" />
	<link rel="alternate" href="https://campusempresa.cat/mod/java/06-02-try-catch-block" hreflang="ca" />
	<link rel="alternate" href="https://enterprisecampus.net/mod/java/06-02-try-catch-block" hreflang="en" />
    
	<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
	<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons/font/bootstrap-icons.css">
	<link href="/css/site.css" rel="stylesheet">
	
    <script src="https://code.jquery.com/jquery-3.7.1.min.js"></script>
  	<script type="text/javascript" src="/js/math_init.js"></script>
  	<script type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/startup.js"></script>
  	<script type="text/javascript" src="/js/cookie.js"></script>
  	<script type="text/javascript" src="/js/main.js"></script>
</head>

<body >
    <div id="content">
		<div id="header" class="container-xxl">
	<div class="row">
		<div class="col-12 col-md-8 p-2 p-md-0">
			<h1 class="m-0 p-0">
				<a href="/"><img src="/img/logo_header.png"></a>
			</h1>
		</div>
		<div class="col-12 col-md-4 p-2 p-md-0 text-end">
			<h2 id="main_title"><cite>Construyendo la sociedad de hoy y del mañana</cite></h2>
			<h3 id="main_subtitle"></h3>
		</div>
	</div>
</div>
<div class="container-xxl" style="margin-top: -1em;">
	<div class="row">
		<div class="col-12 p-2 p-md-0 m-0 text-end">
										<a href="https://enterprisecampus.net/mod/java/06-02-try-catch-block" class="px-2">EN</a></b>
				|
				<b class="px-2">ES</b>
				|
				<a href="https://campusempresa.cat/mod/java/06-02-try-catch-block" class="px-2">CA</a>
								</div>
	</div>
</div>
   <div class="top-bar container-fluid">
	<div class="container-xxl">
		<div class="row">
			<div class="col" id="left_menu">
				<a href="/objective">El Proyecto</a>
				<a href="/about">Sobre nosotros</a>
				<a href="/contribute">Contribuir</a>
				<a href="/donate">Donaciones</a>
				<a href="/licence">Licencia</a>
			</div>
		</div>
	</div>
   </div>

<div class="container-xxl" id="main_content">
	<div class="row">
		<div class="col-12 col-lg-8">
			<div id="nav1" class="navigation"></div>
			<div id="inner_content">
								<div class='row navigation'>
	<div class='col-2'>
					<a href='06-01-introduction-to-exceptions' title="Introducción a las Excepciones">&#x25C4;Anterior</a>
			</div>
	<div class='col-8 text-center'>
					<a href="./"><h2 style="text-decoration:underline">Bloque Try-Catch</h2></a>
			</div>
	<div class='col-2 text-end'>
					<a href='06-03-throw-and-throws' title="Throw y Throws">Siguiente &#x25BA;</a>
			</div>
</div>
<div class='content'><p>El manejo de excepciones es una parte crucial de la programación en Java, ya que permite a los desarrolladores gestionar errores de manera controlada y evitar que las aplicaciones se bloqueen inesperadamente. En esta sección, aprenderemos sobre el bloque <code>try-catch</code>, que es la estructura básica para manejar excepciones en Java.</p>
</div><h1><p>Conceptos Clave</p>
</h1>
<div class='content'><ol>
<li><strong>Excepción</strong>: Un evento que ocurre durante la ejecución de un programa y que interrumpe el flujo normal de las instrucciones.</li>
<li><strong>Bloque <code>try</code></strong>: Contiene el código que puede generar una excepción.</li>
<li><strong>Bloque <code>catch</code></strong>: Contiene el código que maneja la excepción si ocurre.</li>
<li><strong>Bloque <code>finally</code></strong>: (Opcional) Contiene el código que se ejecuta siempre, independientemente de si se lanzó una excepción o no.</li>
</ol>
</div><h1><p>Sintaxis Básica</p>
</h1>
<div class='content'><p>La estructura básica de un bloque <code>try-catch</code> es la siguiente:</p>
</div><div style='position:relative'><a class='copy_button' href='#' onclick='navigator.clipboard.writeText(decodeURIComponent(escape(atob("dHJ5IHsKICAgIC8vIEPDs2RpZ28gcXVlIHB1ZWRlIGxhbnphciB1bmEgZXhjZXBjacOzbgp9IGNhdGNoIChUaXBvRGVFeGNlcGNpb24gZSkgewogICAgLy8gQ8OzZGlnbyBwYXJhIG1hbmVqYXIgbGEgZXhjZXBjacOzbgp9"))));alert("¡Copiado!");return false;'><i class='bi bi-copy'></i></a><pre class='code'>try {
    // C&oacute;digo que puede lanzar una excepci&oacute;n
} catch (TipoDeExcepcion e) {
    // C&oacute;digo para manejar la excepci&oacute;n
}</pre></div><div class='content'></div><h2><p>Ejemplo Práctico</p>
</h2>
<div class='content'><p>Vamos a ver un ejemplo simple donde intentamos dividir un número por cero, lo cual genera una excepción <code>ArithmeticException</code>.</p>
</div><div style='position:relative'><a class='copy_button' href='#' onclick='navigator.clipboard.writeText(decodeURIComponent(escape(atob("cHVibGljIGNsYXNzIFRyeUNhdGNoRXhhbXBsZSB7CiAgICBwdWJsaWMgc3RhdGljIHZvaWQgbWFpbihTdHJpbmdbXSBhcmdzKSB7CiAgICAgICAgdHJ5IHsKICAgICAgICAgICAgaW50IHJlc3VsdCA9IDEwIC8gMDsgLy8gRXN0byBsYW56YXLDoSBBcml0aG1ldGljRXhjZXB0aW9uCiAgICAgICAgICAgIFN5c3RlbS5vdXQucHJpbnRsbigiRWwgcmVzdWx0YWRvIGVzOiAiICsgcmVzdWx0KTsKICAgICAgICB9IGNhdGNoIChBcml0aG1ldGljRXhjZXB0aW9uIGUpIHsKICAgICAgICAgICAgU3lzdGVtLm91dC5wcmludGxuKCJFcnJvcjogTm8gc2UgcHVlZGUgZGl2aWRpciBwb3IgY2Vyby4iKTsKICAgICAgICB9CiAgICB9Cn0="))));alert("¡Copiado!");return false;'><i class='bi bi-copy'></i></a><pre class='code'>public class TryCatchExample {
    public static void main(String[] args) {
        try {
            int result = 10 / 0; // Esto lanzar&aacute; ArithmeticException
            System.out.println(&quot;El resultado es: &quot; + result);
        } catch (ArithmeticException e) {
            System.out.println(&quot;Error: No se puede dividir por cero.&quot;);
        }
    }
}</pre></div><div class='content'></div><h2><p>Explicación del Código</p>
</h2>
<div class='content'><ol>
<li><strong>Bloque <code>try</code></strong>: Contiene el código que intenta dividir 10 por 0.</li>
<li><strong>Bloque <code>catch</code></strong>: Captura la excepción <code>ArithmeticException</code> y ejecuta el código dentro del bloque <code>catch</code>, imprimiendo un mensaje de error.</li>
</ol>
</div><h2><p>Múltiples Bloques <code>catch</code></p>
</h2>
<div class='content'><p>Puedes tener múltiples bloques <code>catch</code> para manejar diferentes tipos de excepciones.</p>
</div><div style='position:relative'><a class='copy_button' href='#' onclick='navigator.clipboard.writeText(decodeURIComponent(escape(atob("cHVibGljIGNsYXNzIE11bHRpcGxlQ2F0Y2hFeGFtcGxlIHsKICAgIHB1YmxpYyBzdGF0aWMgdm9pZCBtYWluKFN0cmluZ1tdIGFyZ3MpIHsKICAgICAgICB0cnkgewogICAgICAgICAgICBpbnRbXSBudW1iZXJzID0gezEsIDIsIDN9OwogICAgICAgICAgICBTeXN0ZW0ub3V0LnByaW50bG4obnVtYmVyc1s1XSk7IC8vIEVzdG8gbGFuemFyw6EgQXJyYXlJbmRleE91dE9mQm91bmRzRXhjZXB0aW9uCiAgICAgICAgfSBjYXRjaCAoQXJpdGhtZXRpY0V4Y2VwdGlvbiBlKSB7CiAgICAgICAgICAgIFN5c3RlbS5vdXQucHJpbnRsbigiRXJyb3I6IE9wZXJhY2nDs24gYXJpdG3DqXRpY2EgaW52w6FsaWRhLiIpOwogICAgICAgIH0gY2F0Y2ggKEFycmF5SW5kZXhPdXRPZkJvdW5kc0V4Y2VwdGlvbiBlKSB7CiAgICAgICAgICAgIFN5c3RlbS5vdXQucHJpbnRsbigiRXJyb3I6IMONbmRpY2UgZGUgYXJyZWdsbyBmdWVyYSBkZSByYW5nby4iKTsKICAgICAgICB9IGNhdGNoIChFeGNlcHRpb24gZSkgewogICAgICAgICAgICBTeXN0ZW0ub3V0LnByaW50bG4oIkVycm9yOiBPY3VycmnDsyB1bmEgZXhjZXBjacOzbi4iKTsKICAgICAgICB9CiAgICB9Cn0="))));alert("¡Copiado!");return false;'><i class='bi bi-copy'></i></a><pre class='code'>public class MultipleCatchExample {
    public static void main(String[] args) {
        try {
            int[] numbers = {1, 2, 3};
            System.out.println(numbers[5]); // Esto lanzar&aacute; ArrayIndexOutOfBoundsException
        } catch (ArithmeticException e) {
            System.out.println(&quot;Error: Operaci&oacute;n aritm&eacute;tica inv&aacute;lida.&quot;);
        } catch (ArrayIndexOutOfBoundsException e) {
            System.out.println(&quot;Error: &Iacute;ndice de arreglo fuera de rango.&quot;);
        } catch (Exception e) {
            System.out.println(&quot;Error: Ocurri&oacute; una excepci&oacute;n.&quot;);
        }
    }
}</pre></div><div class='content'></div><h2><p>Ejercicio Práctico</p>
</h2>
<div class='content'><p><strong>Ejercicio 1: Manejo de Excepciones</strong></p>
<p>Escribe un programa que lea dos números enteros del usuario y los divida. Maneja las posibles excepciones que puedan ocurrir, como <code>ArithmeticException</code> y <code>InputMismatchException</code>.</p>
</div><div style='position:relative'><a class='copy_button' href='#' onclick='navigator.clipboard.writeText(decodeURIComponent(escape(atob("aW1wb3J0IGphdmEudXRpbC5TY2FubmVyOwppbXBvcnQgamF2YS51dGlsLklucHV0TWlzbWF0Y2hFeGNlcHRpb247CgpwdWJsaWMgY2xhc3MgRGl2aXNpb25XaXRoRXhjZXB0aW9uSGFuZGxpbmcgewogICAgcHVibGljIHN0YXRpYyB2b2lkIG1haW4oU3RyaW5nW10gYXJncykgewogICAgICAgIFNjYW5uZXIgc2Nhbm5lciA9IG5ldyBTY2FubmVyKFN5c3RlbS5pbik7CiAgICAgICAgdHJ5IHsKICAgICAgICAgICAgU3lzdGVtLm91dC5wcmludCgiSW5ncmVzZSBlbCBwcmltZXIgbsO6bWVybzogIik7CiAgICAgICAgICAgIGludCBudW0xID0gc2Nhbm5lci5uZXh0SW50KCk7CiAgICAgICAgICAgIFN5c3RlbS5vdXQucHJpbnQoIkluZ3Jlc2UgZWwgc2VndW5kbyBuw7ptZXJvOiAiKTsKICAgICAgICAgICAgaW50IG51bTIgPSBzY2FubmVyLm5leHRJbnQoKTsKICAgICAgICAgICAgaW50IHJlc3VsdCA9IG51bTEgLyBudW0yOwogICAgICAgICAgICBTeXN0ZW0ub3V0LnByaW50bG4oIkVsIHJlc3VsdGFkbyBlczogIiArIHJlc3VsdCk7CiAgICAgICAgfSBjYXRjaCAoQXJpdGhtZXRpY0V4Y2VwdGlvbiBlKSB7CiAgICAgICAgICAgIFN5c3RlbS5vdXQucHJpbnRsbigiRXJyb3I6IE5vIHNlIHB1ZWRlIGRpdmlkaXIgcG9yIGNlcm8uIik7CiAgICAgICAgfSBjYXRjaCAoSW5wdXRNaXNtYXRjaEV4Y2VwdGlvbiBlKSB7CiAgICAgICAgICAgIFN5c3RlbS5vdXQucHJpbnRsbigiRXJyb3I6IEVudHJhZGEgaW52w6FsaWRhLiBQb3IgZmF2b3IsIGluZ3Jlc2UgbsO6bWVyb3MgZW50ZXJvcy4iKTsKICAgICAgICB9IGZpbmFsbHkgewogICAgICAgICAgICBzY2FubmVyLmNsb3NlKCk7CiAgICAgICAgICAgIFN5c3RlbS5vdXQucHJpbnRsbigiRWwgYmxvcXVlIGZpbmFsbHkgc2UgaGEgZWplY3V0YWRvLiIpOwogICAgICAgIH0KICAgIH0KfQ=="))));alert("¡Copiado!");return false;'><i class='bi bi-copy'></i></a><pre class='code'>import java.util.Scanner;
import java.util.InputMismatchException;

public class DivisionWithExceptionHandling {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        try {
            System.out.print(&quot;Ingrese el primer n&uacute;mero: &quot;);
            int num1 = scanner.nextInt();
            System.out.print(&quot;Ingrese el segundo n&uacute;mero: &quot;);
            int num2 = scanner.nextInt();
            int result = num1 / num2;
            System.out.println(&quot;El resultado es: &quot; + result);
        } catch (ArithmeticException e) {
            System.out.println(&quot;Error: No se puede dividir por cero.&quot;);
        } catch (InputMismatchException e) {
            System.out.println(&quot;Error: Entrada inv&aacute;lida. Por favor, ingrese n&uacute;meros enteros.&quot;);
        } finally {
            scanner.close();
            System.out.println(&quot;El bloque finally se ha ejecutado.&quot;);
        }
    }
}</pre></div><div class='content'></div><h2><p>Solución del Ejercicio</p>
</h2>
<div class='content'><ol>
<li><strong>Bloque <code>try</code></strong>: Lee dos números del usuario y los divide.</li>
<li><strong>Bloque <code>catch</code></strong>: Captura <code>ArithmeticException</code> si el segundo número es cero y <code>InputMismatchException</code> si la entrada no es un número entero.</li>
<li><strong>Bloque <code>finally</code></strong>: Cierra el <code>Scanner</code> y ejecuta un mensaje final.</li>
</ol>
</div><h2><p>Retroalimentación sobre Errores Comunes</p>
</h2>
<div class='content'><ul>
<li><strong>Olvidar cerrar recursos</strong>: Siempre cierra recursos como <code>Scanner</code> en un bloque <code>finally</code> para evitar fugas de recursos.</li>
<li><strong>Capturar excepciones genéricas</strong>: Capturar <code>Exception</code> en lugar de excepciones específicas puede ocultar errores y hacer que el código sea más difícil de depurar.</li>
</ul>
</div><h1><p>Conclusión</p>
</h1>
<div class='content'><p>El bloque <code>try-catch</code> es una herramienta esencial para manejar excepciones en Java. Permite a los desarrolladores escribir código más robusto y resistente a errores. En la próxima sección, exploraremos el uso de <code>throw</code> y <code>throws</code> para lanzar y declarar excepciones.</p>
</div><div class='row navigation'>
	<div class='col-2'>
					<a href='06-01-introduction-to-exceptions' title="Introducción a las Excepciones">&#x25C4;Anterior</a>
			</div>
	<div class='col-8 text-center'>
			</div>
	<div class='col-2 text-end'>
					<a href='06-03-throw-and-throws' title="Throw y Throws">Siguiente &#x25BA;</a>
			</div>
</div>

			</div>
		</div>
		<div class="col-12 col-lg-4 publi" id="div_publi">
			<h1>Publicidad</h1>
			<p>Este espacio está destinado a publicidad.</p>
			<p>Si quieres ser patrocinador, contáctanos para incluir enlaces en esta zona: <a href='mailto:admin@campusempresa.cat'>admin@campusempresa.cat</a></p>
			<p>¡Gracias por colaborar!</p>
		</div>
	</div>
</div>

   <div class="container-xxl my-3">
	<div class="row">
		<div class="col">
			<footer>&copy; Copyright 2024. Todos los derechos reservados</footer>
		</div>
	</div>
</div>	

<div id="cookies_adv" style="display:none;">
	Usamos cookies para mejorar tu experiencia de uso y ofrecer contenidos adaptados a tus intereses.
    <a href="#" id="btn_accept_cookies" class="button">Aceptar</a>
    <a href="/cookies">Mas información</a>
</div>	

	</div>    
</body>
</html>
